crates/transaction-generator-lib/src/lib.rs

```diff
@@ -241,24 +260,29 @@ pub async fn create_txn_generator_creator(
                         SamplingMode::Basic,
                     )),
                     *sender_use_account_pool,
-                    accounts_pool.clone(),
+                    &accounts_pool,
                 ),
                 TransactionType::AccountGeneration {
                     add_created_accounts_to_pool,
                     max_account_working_set,
                     creation_balance,
                 } => Box::new(AccountGeneratorCreator::new(
                     txn_factory.clone(),
-                    addresses_pool.clone(),
-                    accounts_pool.clone(),
-                    *add_created_accounts_to_pool,
+                    add_created_accounts_to_pool.then(|| {
```

Comment:
Earlier, we were passing addresses_pool and accounts_pool as input to `AccountGeneratorCreator::new` irrespective of `add_created_accounts_to_pool` flag. Now, we are passing these inputs only if this flag is set to true. Why is that?